tests: logging: log_link_order: Fix uninitialized memory access

nordic-krch opened the pull request:
Mocks used in the test were not initializing certain variables. Valgrind detected that. Test was passing but could fail if some garbage would end up in that memory.

Fixes #57659.

aescolar reviewed (approved):
+1 That fixes the problem.

(I have to recognize that it looks like a weird place to set those values, I couldn't find any docs of that API, but those length and level would seem to be inputs for those calls?)

nordic-krch committed:
tests: logging: log_link_order: Fix uninitialized memory access

Mocks used in the test were not initializing certain variables.
Valgrind detected that. Test was passing but could fail if some
garbage would end up in that memory.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>
